test/Bugs/bug_OS17614914.js

```diff
@@ -0,0 +1,37 @@
+//-------------------------------------------------------------------------------------------------------
+// Copyright (C) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
+//-------------------------------------------------------------------------------------------------------
+
+WScript.LoadScriptFile("..\\UnitTestFramework\\UnitTestFramework.js");
```

Comment:
You test file can be summarized with
```js
Object.prototype.length = undefined;
var ary = Array();
ary.prop1 = 1;
Object.defineProperty(ary, "prop2", {
  value: 1,
  enumerable: false
});
for (var prop in ary) {
  if (prop !== "prop1") {
    console.log(`Fail: ${prop} property should not show in for-in`);
  }
}
console.log("pass");
```

Also, we generally don't use the UnitTestFrameWork for single tests. Since your whole file is related to a single bug and is not likely to have any other cases, just drop the test harness and stick to the test scenario